src/dvc_objects/fs/utils.py

```diff
@@ -189,6 +191,9 @@ def copyfile(
 
     try:
         system.reflink(src, dest)
+        if sys.platform == "darwin":
+            # NOTE: reflink on macos also clones src permissions
```

Comment:
So to be clear, what pyreflink is doing is correct, then, right? It is always preserving permissions. It is not doing any chmod operations on macOS because clonefile does that already, but pyreflink is copying permissions on Linux as ficlone does not do that.

Reply:
So, to be a 1:1 replacement with pyreflink, it has to be:

```python
reflink(src, dst)
if sys.platform == "linux":
    st = os.stat(src)
    os.chown(dst, st.st_uid, st.st_gid)
```

Reply:
I'm not sure I would say that cloning permissions is correct, as it seems inconsistent, a bit wasteful and also kinda unfinished, since it only copies mode but not other stuff (like shutil.copystats). E.g. for us the fact that pyreflink does that on linux is a complete waste, as we need to reset permissions ourselves anyway.

So yeah, we are chmodding different perms and for different reasons. And pyreflink is just trying to be consistent between linux and macos (thouhg I'm not sure if macos clone file clones stats in addition to just mode).